#165372175 Setup Database

nadralia opened the pull request:
**What does this PR do?**

Setup PostgreSQL as database engine


**Description of task to be completed**


Add psycopg2 and python-decouple as application dependencies to requirements.txt
Change default database engine from sqlite3 to PostgreSQL

**How should this be manually tested?**

After cloning the repo, cd into it and run the following commands to activate the virtual environment

$ source venv/bin/activate
Install all requirements

$ pip install -r requirements.txt
Create a .env file the contains your environment values as shown below:
   export DB_NAME=''
   export DB_USER=''
   export DB_PASS=''
   export DB_HOST=''
   export DB_PORT=''

Make migrations

$ python manage.py makemigrations
$ python manage.py migrate
Run the server

$ python manage.py runserver

**Any background context you want to provide?**


The application was setup with sqlite as the default database engine, which does not support hosting on heroku. So now we have sqlite for testing environment and postgres for development/production environment

**What are the relevant pivotal tracker stories?**

#165372175

Reviewers
@kalsmic @bisonlou @AtamaZack @llwasampijja @JamesMudidi

bisonlou commented:
Good work. However, you may need to specify the environmental variables used. Otherwise testing your work takes a long time.

llwasampijja commented:
Great work @nadralia , however, you should ignore ".pyc" files.

JamesMudidi commented on ireporter/settings.py:
```diff
@@ -73,12 +73,37 @@
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
+if os.environ.get("APP_ENV") == "Development":
+    DATABASES = {
```
Having the same code for Development and Production will raise a duplication issue with Maintainability which will give us a rating of C

kalsmic commented:
@nadralia Please update the requirements file

kalsmic commented on requirements.txt:
```diff
@@ -1,4 +1,8 @@
 djangorestframework==3.9.2
+GDAL==2.4.1
+nose==1.3.7
+numpy==1.16.2
```
Please update the requirements file. Packages like numpy and GDAL might not be used in our Django project

bisonlou replied:
I have been able to apply migrations and run the app. Just add psychopg2_binary to the requirements file.

nadralia replied:
@kalsmic yeah am also wondering how they got there

@bisonlou  thats the same thing checkout . https://pypi.org/project/psycopg2-binary/

kalsmic replied:
@nadralia, they might installed system-wide on your computer, this way when you create the virtual environment, system-wide packages come along as well otherwise I do not know how else.

nadralia committed:
Ch(Database Setup): Setup Database
Added .env_sample file
changed the gitignore file
change the database setting in setting.py

[Finishes #165372175]

nadralia committed:
Ch(Database Setup): Setup Database
- Deleted the .pyc files
- Modify settings file

[Finishes #165372175]

bisonlou reviewed (approved):
Great Work

ZackAtama commented:
Great work @nadralia, however, I still see merge conflicts, please resolve them and we have the PR merged.

nadralia committed:
Fixing merge conflicts

nadralia committed:
Fixing merge conflicts and deleting .pyc files plus db.sqlite

nadralia committed:
fixing merge conflicts

nadralia committed:
removed db.sqlite and .pyc

llwasampijja reviewed (approved):
Nice work @nadralia
